domain/src/SharePooling/Services/DisposalProcessor.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace Domain\SharePooling\Services;
+
+use Brick\DateTime\LocalDate;
+use Domain\SharePooling\Actions\DisposeOfSharePoolingToken;
+use Domain\SharePooling\ValueObjects\QuantityBreakdown;
+use Domain\SharePooling\ValueObjects\SharePoolingTokenDisposal;
+use Domain\SharePooling\ValueObjects\SharePoolingTransactions;
+use Domain\ValueObjects\FiatAmount;
+use Domain\ValueObjects\Quantity;
+
+final class DisposalProcessor
```

Comment:
ℹ️ This service essentially calculates the cost basis of a disposal by looking at past and future transactions, following the various share pooling rules (same-day, 30-day, section 104 pool)